torch/_inductor/select_algorithm.py

```diff
@@ -356,6 +369,14 @@ def generate(
         fake_out = ir.Buffer("buf_out", layout)
         kernel_name = f"triton_{self.name}"
 
+        numel = sympy_product(layout.size)
+        buffers = itertools.chain(input_nodes, (fake_out,))
+        index_dtype = (
```

Comment:
for matmuls int64 indexing is usually pretty catastrophic in terms of perf (due to increased register pressure), so going forward it will be better to split matmuls to use int32 indexing always.

Reply:
To be clear,  is this okay for now or would you prefer an assert?

Reply:
Can you check perf of some big matrix? If it's bad I'd prefer an assert/fallback to aten, generating matmuls is purely perf optimization, we can just not do it for any reason.

Reply:
The results were suspiciously similar so I looked at the triton code and of course the templates hard-code 32-bit indexing which in hindsight is obvious. I've changed it to not add the kernel as a choice to the autotuner when I detect 32-bit indexing would overflow.